core/storage/trie/polkadot_trie_db/polkadot_trie_batch.cpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/trie/polkadot_trie_db/polkadot_trie_batch.hpp"
+
+namespace kagome::storage::trie {
+
+  using common::Buffer;
+
+  PolkadotTrieBatch::PolkadotTrieBatch(PolkadotTrieDb &trie) : trie_{trie} {}
+
+  outcome::result<void> PolkadotTrieBatch::put(const Buffer &key,
```

Comment:
maybe better take by value and then move?
```suggestion
  outcome::result<void> PolkadotTrieBatch::put(Buffer key,
```

Reply:
pls. refactor

Reply:
It's WriteableMap interface and I'm afraid it's no good changing it to pass-by-value. But you're right that for TrieBatch it would be better. I will think of how to do it.

Reply:
I extended the interface.